pkgs/by-name/zx/zxing/package.nix

```diff
@@ -4,30 +4,45 @@
   fetchurl,
   jre,
 }:
-stdenv.mkDerivation rec {
-  pname = "zxing";
-  version = "3.1.0";
+
+let
+  version = "3.5.3";
+
   # Maven builds are hard to get right
   core_jar = fetchurl {
-    url = "http://repo1.maven.org/maven2/com/google/zxing/core/${version}/core-${version}.jar";
-    sha256 = "199l4xvlcpafqn69r3k9qjpqkw9lvkl02hzpm0ackvdhl7vk42zh";
+    url = "https://repo1.maven.org/maven2/com/google/zxing/core/${version}/core-${version}.jar";
+    hash = "sha256-jYBkwWNv2u9xid2QVcfVmVColAoS8ik5VkRuw8EJ/YI=";
   };
+
   javase_jar = fetchurl {
-    url = "http://repo1.maven.org/maven2/com/google/zxing/javase/${version}/javase-${version}.jar";
-    sha256 = "0fzxvvf5dqyrs5m9rqw4ffm9h1s27bi7q3jb1dam34s80q2rp2zq";
+    url = "https://repo1.maven.org/maven2/com/google/zxing/javase/${version}/javase-${version}.jar";
+    hash = "sha256-1leVt4nvrIohVssqS2SNIOWL/gNhMWW7Xc5ocOzveY4=";
   };
-  inherit jre;
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "zxing";
+  inherit version jre;
+
   dontUnpack = true;
+
   installPhase = ''
-    mkdir -p "$out/lib/java" "$out/bin"
-    cp "${core_jar}" "${javase_jar}" "$out/lib/java"
-    substituteAll "${./java-zxing.sh}" "$out/bin/java-zxing"
-    substituteAll "${./zxing-cmdline-runner.sh}" "$out/bin/zxing-cmdline-runner"
-    substituteAll "${./zxing-cmdline-encoder.sh}" "$out/bin/zxing-cmdline-encoder"
-    substituteAll "${./zxing.sh}" "$out/bin/zxing"
-    chmod a+x "$out/bin"/*
-    cd "$out/lib/java"; for i in *.jar; do mv "$i" "''${i#*-}"; done
+    runHook preInstall
+
+    mkdir -p $out/lib/java $out/bin
```

Comment:
I honestly find fully quoted variable expansions more readable in general (yes, our stdenv is messed up and we cannot yet afford spaces in the store path, but unquoted expansions should always cause stopping and thinking about the context)